rossum_api/api_client.py

```diff
@@ -81,6 +82,50 @@ async def authenticate_if_needed(self: APIClient, *args, **kwargs):
     return authenticate_if_needed
 
 
+class Retry:
```

Comment:
Do we really need a class for this? It sounds too heavy for me. In my ideal world, we can either have a (factory) function, which gives us the decorator or we can implement the retry directly in the `_request`. 

Second option is maybe even better as decorator usage feels very artificial to me in this case.

Reply:
Agreed. Class sounds like an overkill to me.

One more thing. Should we also add some **jitter** to spread the traffic of retried requests from different applications in time? Some arguments why this is a good idea can be find here: https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/

Reply:
Agree that class is an overkill. Implemented directly, the logic is IMHO a bit mixed but nothing crucial. Jitter 👍